enhancements/ingress/set-delete-http-headers.md

```diff
@@ -1123,6 +1138,33 @@ We have multiple options for quoting header values in `haproxy.config`, as [desc
   2. Wrap the entire string in single-quotes.
 The HAProxy documentation refers to this as _strong quoting_.  Using strong quoting allows us to keep the quoting logic simple and `haproxy.config` more human-readable compared to the alternatives.
 
+### Proposed Alternative Solution for overriding the HTTP response headers defined in the IngressController by the Route
+As per HAProxy's design: the frontend receives the request and dispatches it to the backend, and then the frontend
+gets the response from the backend.
+
+So, the override behavior happens the way mentioned in the question `Why does a custom header value defined in RouteSpec override IngressControllerSpec value ?`
+under section `Open Questions` where HTTP Response headers defined in the IngressController are overridden by headers 
+defined in the Route.
+
+However, if we want the same behavior of override for HTTP response headers as that of HTTP request headers then we will have
+to inject the set-header from the IngressController config into every backend and not the frontend.
+
+The downside is that it would grow the generated haproxy.config significantly. 
+
+The upside is:
+
+It would give us better control over the ordering.
+
+It would eliminate the issue with the host header. So, we could allow it to be set via IngressController.
+
+The reason for this proposal was that it's quite common to have a global setting with an escape hatch locally to 
+override the global setting.
+The purpose of this EP is to define custom headers. So, there are two reasons we need such override:
+1. A route owner shall have the choice to define what headers he wants as per his application requirements(as they might be varying)
+   than the ones admin has set cluster-wide.
+2. The IngressController and Route might be owned by the same person who wants to specify some global configuration
+   and then override that global configuration on a small subset of routes.
+
```

Comment:
This doesn't belong here, it doesn't add to the meaning.

Reply:
it does not exist now.